### DISCUSSION TITLE
What are good learning resources for people interested in timeseries?

### DISCUSSION THREAD

**Discussion by microprediction** [microprediction/timemachines#127] [General]
Here's what ChatGPT suggests:

**Comment by microprediction** [comment]
There are many excellent books available on the topic of time series modeling. Some of the best include:

- "Forecasting: Principles and Practice" by Rob J Hyndman and George Athanasopoulos: This book provides a comprehensive introduction to time series modeling, including both theoretical and practical aspects. It covers a wide range of topics, including basic time series analysis, statistical forecasting methods, and advanced topics such as seasonal decomposition and exponential smoothing.

- "Time Series Analysis and Its Applications" by Robert H. Shumway and David S. Stoffer: This book is a popular textbook on time series analysis that covers a wide range of topics, including ARIMA models, stationary and nonstationary time series, and forecasting. It includes numerous examples and exercises to help you apply the concepts to real-world data.

- "Time Series Analysis and Forecasting by Example" by Demetris T. Christodoulou and Diogo Silva: This book provides a practical introduction to time series analysis and forecasting using Python. It covers a wide range of topics, including time series data manipulation, visualization, and modeling, and includes hands-on examples and code snippets to help you learn the concepts.

Overall, I would recommend starting with "Forecasting: Principles and Practice" by Hyndman and Athanasopoulos, as it provides a broad overview of the field and covers both the theoretical and practical aspects of time series modeling. Once you have a solid foundation in the basics, you can move on to more specialized books, such as "Time Series Analysis and Its Applications" by Shumway and Stoffer or "Time Series Analysis and Forecasting by Example" by Christodoulou and Silva.

**Comment by microprediction** [comment]
Ram likes:
 - https://www.linkedin.com/posts/ram-seshadri-nyc-nj_ml-python-finance-activity-7028939156198170624-ddU0?utm_source=share&utm_medium=member_desktop

**Comment by microprediction** [comment]
As of my knowledge cutoff date in September 2021, there are several good books that cover time series analysis and forecasting using Python. Here are a few notable ones:

- "Practical Time Series Analysis: Prediction with Statistics and Machine Learning" by Dr. Aileen Nielsen

This book provides a practical introduction to time series analysis and forecasting. It includes hands-on examples with Python code and covers both classical statistical techniques and more modern machine learning methods.

- "Python for Time Series Data Analysis" by Federico Marani

This book provides an introduction to time series data analysis using Python. It covers important topics like data wrangling, exploratory data analysis, time series decomposition, and forecasting. The book also introduces various machine learning techniques for time series prediction.


- "Time Series Analysis and Forecasting with Python" by Avishek Pal and PKS Prakash

This book covers time series analysis and forecasting using Python. It includes both classical methods (such as ARIMA) and machine learning approaches (such as LSTM). It also provides a practical guide to applying these techniques to real-world datasets.


- "Hands-On Time Series Analysis with R and Python" by Rami Krispin

This book provides a comprehensive introduction to time series analysis and forecasting using both R and Python. It covers a wide range of techniques, including moving average models, autoregressive models, and machine learning methods. The book also includes practical case studies.

- "Introduction to Time Series Forecasting with Python: How to Prepare Data and Develop Models to Predict the Future" by Jason Brownlee

This book provides a detailed introduction to time series forecasting with Python. It covers the entire forecasting process, from data preparation to model evaluation. It also includes a range of practical examples and case studies.

**Comment by microprediction** [comment]
A free course here
https://www.forecaster.guru/time-series-analytics-and-forecasting.html

**Comment by pcotton-intech** [comment]
Peter likes:

  * Harrison and West
  * ...

**Reply by MarkBrezina** [reply]
https://link.springer.com/book/10.1007/b98971
